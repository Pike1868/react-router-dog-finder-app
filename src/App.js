import "./App.css";
import dogsArr from "./dogs";
import RouteList from "./RouteList";
import NavBar from "./NavBar";

/**
 * App Component:
 * 
 * Top level component for the DogFinder application. Sets up the main structure of the application,
 * including the heading, the NavBar for navigation, and the RouteList which defines the routing logic. 
 */

function App() {
  return (
    <div className="App">
      <h1>Welcome to Dog Finder!</h1>
      <NavBar dogs={dogsArr} />
      <RouteList dogs={dogsArr} />
    </div>
  );
}

export default App;
