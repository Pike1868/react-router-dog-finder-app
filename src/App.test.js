import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import App from "./App";

//Smoke Test
test("App component renders without crashing", () => {
  render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
});

//Snapshot Test
test("App component matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});


describe("App Component Tests", () => {
  // Common setup (e.g., rendering)
  beforeEach(() => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
  });

  test("displays the correct heading", () => {
    const appHeading = screen.getByText("Welcome to Dog Finder!");
    expect(appHeading).toBeInTheDocument();
  });

  test("displays the NavBar", () => {
    const navElement = screen.getByRole('navigation');
    expect(navElement).toBeInTheDocument();
  });

});