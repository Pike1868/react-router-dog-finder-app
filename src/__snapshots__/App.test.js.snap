// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Welcome to Dog Finder!
    </h1>
    <nav
      style="display: flex; justify-content: space-around; background-color: lightblue; padding: 10px;"
    >
      <a
        aria-current="page"
        class="active"
        href="/dogs"
      >
        Home
      </a>
      <a
        class=""
        href="/dogs/whiskey"
      >
        Whiskey
      </a>
      <a
        class=""
        href="/dogs/duke"
      >
        Duke
      </a>
      <a
        class=""
        href="/dogs/perry"
      >
        Perry
      </a>
      <a
        class=""
        href="/dogs/tubby"
      >
        Tubby
      </a>
    </nav>
    <div
      class="DogList"
    >
      <div
        class="row mt-4"
      >
        <div
          class="col"
        >
          <h2
            class="text-center"
          >
            HELLO. WE HAVE DOGS. CLICK ON THEM FOR MORE INFO.
          </h2>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-3 text-center"
        >
          <img
            alt="Whiskey"
            src="/whiskey.jpg"
          />
          <h3
            class="mt-3"
          >
            <a
              href="/dogs/whiskey"
            >
              Whiskey
            </a>
          </h3>
        </div>
        <div
          class="col-3 text-center"
        >
          <img
            alt="Duke"
            src="/duke.jpg"
          />
          <h3
            class="mt-3"
          >
            <a
              href="/dogs/duke"
            >
              Duke
            </a>
          </h3>
        </div>
        <div
          class="col-3 text-center"
        >
          <img
            alt="Perry"
            src="/perry.jpg"
          />
          <h3
            class="mt-3"
          >
            <a
              href="/dogs/perry"
            >
              Perry
            </a>
          </h3>
        </div>
        <div
          class="col-3 text-center"
        >
          <img
            alt="Tubby"
            src="/tubby.jpg"
          />
          <h3
            class="mt-3"
          >
            <a
              href="/dogs/tubby"
            >
              Tubby
            </a>
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
